demo/lightning/text-based/model.py

```diff
@@ -0,0 +1,227 @@
+        super().__init__(*args, **kwargs)
+        self.columns = columns
+        self.batch_size = batch_size
+        self.current_file = None
+        self.columns = columns
+        """
+        A sublcass of the DatafluxIterableDataset, this dataset allows for the reading and batch
+        distribution of a parquet file, where batch-size corresponds to rows of the parquet table.
+        """
+
+    def __iter__(self):
+        """
+        Overriding the __iter__ function allows batch size to be used to sub-divide the contents of
+        individual parquet files.
+        """
+        worker_info = data.get_worker_info()
+        if worker_info is None:
+            print("Single-process data loading detected", flush=True)
+            for bytes_content in df_iter.dataflux_core.download.dataflux_download_lazy(
+                    project_name=self.project_name,
+                    bucket_name=self.bucket_name,
+                    objects=self.objects,
+                    storage_client=self.storage_client,
+                    dataflux_download_optimization_params=self.
+                    dataflux_download_optimization_params,
+                    retry_config=self.config.download_retry_config,
+            ):
+                table = self.data_format_fn(bytes_content)
+                for batch in table.iter_batches(batch_size=self.batch_size,
+                                                columns=self.columns):
+                    yield from batch.to_pylist()
+        else:
+            # Multi-process data loading. Split the workload among workers.
+            # Ref: https://pytorch.org/docs/stable/data.html#torch.utils.data.IterableDataset.
+            per_worker = int(
+                math.ceil(len(self.objects) / float(worker_info.num_workers)))
+            worker_id = worker_info.id
+            start = worker_id * per_worker
+            end = min(start + per_worker, len(self.objects))
+            for bytes_content in df_iter.dataflux_core.download.dataflux_download_lazy(
+                    project_name=self.project_name,
+                    bucket_name=self.bucket_name,
+                    objects=self.objects[start:end],
+                    storage_client=self.storage_client,
+                    dataflux_download_optimization_params=self.
+                    dataflux_download_optimization_params,
+                    retry_config=self.config.download_retry_config,
+            ):
+                table = self.data_format_fn(bytes_content)
+                for batch in table.iter_batches(batch_size=self.batch_size,
+                                                columns=self.columns):
+                    yield from batch.to_pylist()
+
+
+def main():
+    args = parse_args()
+    logging.basicConfig(level=args.log_level)
+    # The prefix passed here determines which parquet files are read from our bucket.
+    # config = df_iter.Config(prefix="fineweb/sample/10BT/014_00000.parquet")
```

Comment:
Did you mean to leave this commented out?

Reply:
Nope, will remove